### PR TITLE
Responsive styling

### LINKED ISSUE #357: Responsive layouts
While I'm still not quite happy with the state of layout abstractions, I think it's time to ensure the site works well on mobile devices.

Once we know what the relevant breakpoints are we need to update the pre-renderer so it injects the critical CSS for the respective breakpoints - see #457.

### TODO

- [ ] the shapes are displaced on some mobiles (seems to be the case on iOS but also when simulating a Galaxy S5 in Chrome): 

Bildschirmfoto 2019-05-25 um 13 02 24

- [ ] the author badge on mobile is currently hidden and needs to be shown again and fixed so it renders correctly/reasonably on mobile
- [ ] intermediate resolutions between desktop and mobile (e.g. iPad) are currently broken and need to be addressed as well, e.g.:

Bildschirmfoto 2019-05-25 um 13 08 04

- [ ] some of the shapes have excessive amounts of whitespace in some resolutions:

Bildschirmfoto 2019-05-27 um 14 40 33

- [ ] we should restore the parallax scrolling effects for the shapes on the desktop

### PR DESCRIPTION
**Pull request by pichfl**
- [x] Responsive shapes for Home
- [x] Responsive shapes for Services
- [x] Responsive shapes for Work
- [x] Responsive shapes for Why
- [x] Responsive shapes for Contact
- [x] Vertical spacing
- [x] Ensure ShapeFeature works in all variations

From @marcoow  https://github.com/simplabs/simplabs.github.io/pull/500#pullrequestreview-241298864

- [x] Blog shape
- [x] Full Stack Engineering page vertical offset
- [x] the team augmentation page vertical offset
- [x] tutoring page vertical offsets
- [x] expedition case vertical offset
- [x] timify case vertical offset
- [x] Work page shape tagline off
- [x] Do not show video on mobile, fall back to static image
- [x] elixir page the EEF sponsor image

addresses #357

### CODE REVIEW

**Review by marcoow — COMMENTED** [review]
I think this works pretty well for the homepage already but there are some minor things that I recognized:

* the trainline logo overlaps with the devices for the trainline case - we can probably hide the logo altogether on mobile?

Bildschirmfoto 2019-05-21 um 20 40 54

* the card image for the upcoming events card isn't tall enough:

Bildschirmfoto 2019-05-21 um 20 41 02

* there's a gap between the 2 shapes of the Ember experts teaser:

Bildschirmfoto 2019-05-21 um 20 41 06

**Comment by ghislaineguerin** [comment]
@pichfl Is there a way to scale down spacing in general when on mobile? I feel it's a bit disconnected

IMG_5020
IMG_5021

**Comment by pichfl** [comment]
Sure! This is only a draft after all, there is still a lot to do to make this look right.

**Review by marcoow — COMMENTED** [review]
I found a few relatively small things:

* the shape at the top of the blog makes the menu toggle invisible: 
Bildschirmfoto 2019-05-23 um 17 46 20
* on the work page, the label on the trainline case teaser shows but is only partly visible: 
Bildschirmfoto 2019-05-23 um 17 49 38
* on the work page there seems to be a bit too much space between the timify case teaser and the content below it: 
Bildschirmfoto 2019-05-23 um 17 49 58
* on the full stack engineering page there seems to be a bit too much space between the last content paragraph and the playbook teaser: 
Bildschirmfoto 2019-05-23 um 17 51 12
* on the team augmentation page there seems to be a bit too much space between the last content paragraph and the playbook teaser: 
Bildschirmfoto 2019-05-23 um 17 51 12
* on the tutoring page there seems to be a bit too much space between the last content paragraph and the ember teaser: 
Bildschirmfoto 2019-05-23 um 17 52 21
* the video doesn't look correct on mobile but we should show a static image instead there anyway (and ideally make sure the browser doesn't even load the video): 
Bildschirmfoto 2019-05-23 um 17 52 58
* on the expedition case there is a bit too much space between the last content paragraph and the playbook teaser: 
Bildschirmfoto 2019-05-23 um 17 54 04
* on the timify case there is a bit too much space between the last content paragraph and the playbook teaser: 
Bildschirmfoto 2019-05-23 um 17 54 47
* on the elixir page the EEF sponsor image doesn't show up correctly: 
Bildschirmfoto 2019-05-23 um 17 55 19

**Comment by pichfl** [comment]
@marcoow thanks, have added every item as TODO to the description of this PR and will take them on tomorrow.

**Comment by ghislaineguerin** [comment]
@pichfl mobile looks good, besides the spacing issues. However, on a tablet it won't look good. Are we including tablet resolution to the responsive solution?

**Comment by marcoow** [comment]
Yeah, good point @ghislaineguerin - intermediate solutions don't work for everything yet:

Bildschirmfoto 2019-05-24 um 08 51 06
Bildschirmfoto 2019-05-24 um 08 51 28
Bildschirmfoto 2019-05-24 um 08 52 23
Bildschirmfoto 2019-05-24 um 08 52 39
Bildschirmfoto 2019-05-24 um 08 52 59
Bildschirmfoto 2019-05-24 um 08 53 07
Bildschirmfoto 2019-05-24 um 08 53 22
Bildschirmfoto 2019-05-24 um 08 53 30
Bildschirmfoto 2019-05-24 um 08 53 36

I guess we need another breakpoint?

**Comment by marcoow** [comment]
Also on iPad the images in the shapes seem to be displaced:

Bildschirmfoto 2019-05-24 um 08 57 33
Bildschirmfoto 2019-05-24 um 08 57 51

**Comment by pichfl** [comment]
As mentioned in the update call, the medium size will be tackled in a separate PR as this one is already quite large.

File: src/ui/components/ShapeFeature/stylesheet.css [hunk 1]
```diff
@@ -5,10 +5,13 @@
...
-  padding: 28rem 0 14rem;
-  min-height: 58rem;
+  margin: 0 0 16rem;
+  padding: 0;
 
-  --shape-color: var(--link-color);
+  --shape-color: var(--color-accent);
+}
+
+:scope[state|hasVideo] {
```
Comment (marcoow) [inline src/ui/components/ShapeFeature/stylesheet.css]: Is this needed?

**Review by marcoow — COMMENTED** [review]
There are some problems on the desktop now:

* the background images in the shapes are now too small: 
Bildschirmfoto 2019-05-24 um 16 40 46
Bildschirmfoto 2019-05-24 um 16 41 18
Bildschirmfoto 2019-05-24 um 16 41 34
* the video doesn't show/play at all anymore
* the little badges for the case teasers are gone: 
Bildschirmfoto 2019-05-24 um 16 42 18

I'll look at mobile now.

**Review by marcoow — COMMENTED** [review]
* the blog doesn't work anymore with the new author badge: 
deploy-preview-500--gallant-hawking-42050a netlify com_
* the shapes are displaced on the success stories on ios: 
Bildschirmfoto 2019-05-24 um 16 42 18

**Pull request merged**